### DISCUSSION TITLE
Add search box to /sites

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#1049] [Anything Else]
"A quick note on the paginated /sites - I see the reasoning behind it, however it makes getting to a particular site much slower as I have three pages and it takes a second or two for each page to load. Before, I used "Ctrl + F" to find the site I'm looking for and it took a second. It would be great to have a quick search option at the top or even an option to list them all on one page? Is this possible by modifying the URL query parameter - https://plausible.io/sites?page=3"

(feedback via email)

**Comment by ukutaht** [comment]
Makes sense to add this.

It's possible to get all the sites on one page by appending a query parameter `per_page=100`

**Reply by metmarkosaric** [reply]
thanks! that little hack will work for now!

**Comment by andreamazza89** [comment]
Hi 👋  - I'm looking to work on this, how about:

1. We only show a search box if there are more than one page worth of sites. In other words, with 18 sites per page, we only show a search box when there are 19 sites or more

2. Visually, how do we feel about something like this: 
<img width="1129" alt="Screenshot 2022-03-30 at 22 17 53" src="https://user-images.githubusercontent.com/16692471/160924150-e58ecf3d-5668-40dc-b4c4-0774641d85c9.png">

3. We use query parameters to search, for example `plausible.io/sites?search=sible` shows only sites containing 'sible' in their domain name

4. Rather than having the user hit a 'search' button, we wait a few milliseconds after the user stops typing and reload the page with the search term entered (I am assuming this is easy enough to achieve but will need to check)

**Reply by metmarkosaric** [reply]
Nice! Sounds good to me @andreamazza89! To fit it on a mobile it would probably need to be a search icon and when you press it then it becomes what you have now. Also perhaps better to place it on the right side next to the "Add a website" button. We should also shorten the "Add a website" to "Add site" to make it all fit better

**Reply by ukutaht** [reply]
Agree that instead of having a button it should search as you type. Something like this feels good:

https://user-images.githubusercontent.com/3731516/160985921-e7018c12-42f0-4c57-86eb-518f12c7e07c.mp4

**Comment by ukutaht** [comment]
Some implementation notes:

1. To have a smooth user experience, the interaction must be controlled by Javascript on the frontend without reloading the whole HTML page
2. Outside of the dashboard which uses React, we use Alpine.Js for these kinds of things. Example: https://github.com/plausible/analytics/blob/master/lib/plausible_web/templates/billing/upgrade_to_plan.html.eex
3. There already is an endpoint `/api/sites` that is used internally to fetch the list of sites in JSON. It shouldn't be hard to extend that endpoint with a `{query: 'asd'}` or `{search: 'asd'}` parameter.
4. Not 100% on this but I would look into rendering the list completely on the frontend using Alpine.js, even for the first render. If we want a smooth experience like in the screen recording I shared above, we have to have the capability of fetching sites in JSON and rendering the list on the frontend anyways. So why maintain two ways of doing it? Better to use frontend rendering every time.

Do these notes make sense @andreamazza89? Let me know if something is unclear.

**Reply by andreamazza89** [reply]
All clear and agree 100% - Apline.Js looks pretty cool

Thank you 😄

**Comment by metmarkosaric** [comment]
**Update:** We’ve made some improvements to the “My Sites” page:

* You can now search for any of your sites
* There’s a graph for an overview of the traffic 
* And a comparison to the previous period so you can see the trend

Have fun exploring!
